diff --git a/gtsam/CMakeLists.txt b/gtsam/CMakeLists.txt
index 11ae113b9..510d7285f 100644
--- a/gtsam/CMakeLists.txt
+++ b/gtsam/CMakeLists.txt
@@ -155,12 +155,12 @@ target_include_directories(gtsam SYSTEM BEFORE PUBLIC
   $<INSTALL_INTERFACE:include/gtsam/3rdparty/CCOLAMD>
 )
 if(GTSAM_SUPPORT_NESTED_DISSECTION)
-  target_include_directories(gtsam BEFORE PUBLIC
-    $<BUILD_INTERFACE:${GTSAM_SOURCE_DIR}/gtsam/3rdparty/metis/include>
-    $<BUILD_INTERFACE:${GTSAM_SOURCE_DIR}/gtsam/3rdparty/metis/libmetis>
-    $<BUILD_INTERFACE:${GTSAM_SOURCE_DIR}/gtsam/3rdparty/metis/GKlib>
-    $<INSTALL_INTERFACE:include/gtsam/3rdparty/metis/>
-  )
+  find_package(metis_catkin REQUIRED)
+  set(METIS_INCLUDE_DIRECTORIES ${metis_catkin_INCLUDE_DIRS})
+  set(METIS_LIBRARIES ${metis_catkin_LIBRARIES})
+  message("metis catkin libs: ${metis_catkin_LIBRARIES}")
+  add_library(metis SHARED IMPORTED)
+  set_property(TARGET metis PROPERTY IMPORTED_LOCATION ${metis_catkin_LIBRARIES})
 endif()
 
 
